### PR TITLE
Register and document cluster-autoscaler `safe-to-evict` annotation

### LINKED ISSUE #35821: Register and document annotation `cluster-autoscaler.kubernetes.io/safe-to-evict`
**This is a Bug Report**

**Problem:**
The cluster autoscaler uses an annotation, `cluster-autoscaler.kubernetes.io/safe-to-evict`. However, we don't list that annotation in 
[Well-Known Labels, Annotations and Taints](
https://kubernetes.io/docs/reference/labels-annotations-taints/).

**Proposed Solution:**
Update [Well-Known Labels, Annotations and Taints](
https://kubernetes.io/docs/reference/labels-annotations-taints/) to mention the Pod annotation `cluster-autoscaler.kubernetes.io/safe-to-evict` and why you might use it.

**Page to Update:**
https://kubernetes.io/docs/reference/labels-annotations-taints/

**Additional Information:**
/language en

### PR DESCRIPTION
**Pull request by mk46**
Fixes #35821
/language en
/kind bug
/sig autoscaling

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -85,6 +85,14 @@ The current version of the application (e.g., a semantic version, revision hash,
 
 One of the [recommended labels](/docs/concepts/overview/working-with-objects/common-labels/#labels).
 
+### cluster-autoscaler.kubernetes.io/safe-to-evict
+
+Example: `cluster-autoscaler.kubernetes.io/safe-to-evict: true`
+
+Used on: Pod
+
+When this annotation is set to `true`, Pods can be evicted even if they don't meet other requirements for it.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:94]:
```suggestion
When this annotation is set to `true`, the cluster autoscaler is allowed to evict a Pod
even if other rules would normally prevent that.
```

What about if you set it to `false`? We could also explain that if you don't set this annotation then the cluster autoscaler follows its overall behavior.

Comment (mk46) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Done

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -85,6 +85,16 @@ The current version of the application (e.g., a semantic version, revision hash,
...
 
+### cluster-autoscaler.kubernetes.io/safe-to-evict
+
+Example: `cluster-autoscaler.kubernetes.io/safe-to-evict: true`
+
+Used on: Pod
+
+When this annotation is set to `true`, the cluster autoscaler is allowed to evict a Pod
+even if other rules would normally prevent that. If this annotation is set to `false`, Pods can prevent the cluster autoscaler from removing a node.
+If this annotation is not set then the cluster autoscaler follows its overall behavior.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:96]:
```suggestion
When this annotation is set to `true`, the cluster autoscaler is allowed to evict a Pod
even if other rules would normally prevent that.
The cluster autoscaler never evicts Pods that have this annotation explicitly set to
`false`; you could set that on an important Pod that you want to keep running.
If this annotation is not set then the cluster autoscaler follows its overall behavior.
```

**Review by sftim — COMMENTED** [review]
I suggest one more change - talking about the Pod-level impact rather than the overall effect.

**Comment by sftim** [comment]
/retitle Register and document cluster-autoscaler `safe-to-evict` annotation

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (mk46) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Done.

**Comment by mk46** [comment]
@sftim PTAL

**Comment by sftim** [comment]
I'm happy for any contributor to pick up reviewing this.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -85,6 +85,18 @@ The current version of the application (e.g., a semantic version, revision hash,
 
 One of the [recommended labels](/docs/concepts/overview/working-with-objects/common-labels/#labels).
 
+### cluster-autoscaler.kubernetes.io/safe-to-evict
+
+Example: `cluster-autoscaler.kubernetes.io/safe-to-evict: true`
+
+Used on: Pod
+
+When this annotation is set to `true`, the cluster autoscaler is allowed to evict a Pod
```
[the remark this answers is not in the document]
Comment (mk46) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Addressed your suggestion.

**Review by sftim — COMMENTED** [review]
/approve

**Pull request merged**